sympy/functions/elementary/exponential.py

```diff
@@ -1005,28 +1010,29 @@ def coeff_exp(term, x):
 
         # TODO new and probably slow
         try:
-            a, b = arg.leadterm(x)
-            s = arg.nseries(x, n=n+b, logx=logx)
+            a, b = z.leadterm(t, cdir=1)
+            s = z._eval_nseries(t, n=n+b, logx=logx, cdir=1)
         except (ValueError, NotImplementedError, PoleError):
-            s = arg.nseries(x, n=n, logx=logx)
+            s = z._eval_nseries(t, n=n, logx=logx, cdir=1)
             while s.is_Order:
                 n += 1
-                s = arg.nseries(x, n=n, logx=logx)
-        if _logx and logx.has(x):
-            a, b = arg.as_leading_term(x, logx=logx), S.Zero
+                s = z._eval_nseries(t, n=n, logx=logx, cdir=1)
+        if _logx and logx.has(t):
```

Comment:
When does logx have t?

Reply:
No it doesn't and also there's no use of `_logx` now , this check would have come in play if we would have not used the sustitution `z.subs(x, cdir*t)` and whenver `log(x)` would be `x` rather than other symbols like `p`

Reply:
In the last two commits , I've removed whatever code wouldn't come into play and also added some spacing to keep blocks responsible for similar operations together as the code overall was looking to clumsy !